Hani Alfiyyah Nurulhuda

haniialfiyyaah opened the pull request:
# Server:
- [x] API Docs

### Todo REST API:
- [x] Create Todo
- [x] Get All Todo
- [x] Get Todo By Id
- [x] Update Todo
- [x] Update Status Todo
- [x] Delete Todo

### Auth
- [x] Register
- [x] Login
- [x] Middleware Authentication
- [x] Middleware Authorization
- [x] Error Handling
- [x] 3rd Party APIs (Calenderific to get Holiday date)

# Client
- [x] Login
- [x] Register
- [x] CRUD Todo
- [x] 3rdAPI show holiday

.env
SECRET_KEY=haniisecretkey
CALENDARIFIC_API_KEY
GOOGLE_CLIENT_ID

Kendala: front-end

haniialfiyyaah committed:
Initial server with routes

haniialfiyyaah committed:
Sequelize init and Create Todo

haniialfiyyaah committed:
Get todo, update todo, update todo status, delete todo

haniialfiyyaah committed:
Fix status message on todoController

Sursev07 commented on server/controllers/homeController.js:
```diff
@@ -0,0 +1,7 @@
+class Controller {
```
ini ga dipakai kan ya?

haniialfiyyaah replied:
dipake di _routes/index.js_ ka

haniialfiyyaah committed:
Update api_doc and todoController

haniialfiyyaah committed:
1. Validation
2. Migrate Table user dan column UserId
3. Associate User and Todo
4. Add Register and Login
5. Add JWT
6. Add middleware authentication and authorization

haniialfiyyaah committed:
Error Handling

haniialfiyyaah committed:
add 3rd party calendarific to get holiday

haniialfiyyaah committed:
Add google login and 3rdAPI

haniialfiyyaah committed:
add name column

haniialfiyyaah committed:
fix bug google login

haniialfiyyaah committed:
Login and register client

haniialfiyyaah committed:
Clean Code
1. Fix up function and jquery
2. Change index.html
3. delete unused code

haniialfiyyaah committed:
crud todo client

haniialfiyyaah committed:
add 3rd API holiday calendar to client

haniialfiyyaah committed:
add route /holiday to doc

haniialfiyyaah committed:
fix typo update status client

haniialfiyyaah committed:
remove unused code

haniialfiyyaah committed:
fix server response

haniialfiyyaah committed:
deploy

haniialfiyyaah committed:
fix deploy

haniialfiyyaah committed:
fix register client and clean some code

haniialfiyyaah committed:
delete show modal loading
